src/mattack_actors.cpp

```diff
@@ -294,7 +295,11 @@ void bite_actor::load_internal( JsonObject &obj, const std::string &src )
 void bite_actor::on_damage( monster &z, Creature &target, dealt_damage_instance &dealt ) const
 {
     melee_actor::on_damage( z, target, dealt );
-    if( one_in( no_infection_chance - dealt.total_damage() ) ) {
+    bool infectionchance = one_in( no_infection_chance - dealt.total_damage() );
+    //disable biting on ungrabbed targets
+    if( !target.has_effect( effect_grabbed ) ) {
+        infectionchance = 0;
```

Comment:
`infectionchance` is declared as a `bool`, don't assign an integer to it, assign an boolean value (`true` or `false`) to it.

Also: the `else if` branch makes setting `infectionchance` here useless, it is never read after that assignment, so the assignment does not matter.

Also: the name is odd, it's not an "infection chance" (that would be some numeric value, a "X in Y" value), but this is more a "should infection actual happen" value. I suggest a simpler version: `if( has_effect( ... ) && one_in( ... ) ) {` where you don't even need the variable.

Reply:
there was some weirdness i was stumbling into early on in doing this little PoC where I didn't notice that one_in was returning a bool and I was also juggling trying to figure out why there are three enemies that have specific infection chance fields in the json when the rest don't, and I believe the former is the cause for the weirdness in typing.

This is definitely an improvement on what I did, though.